### PR TITLE
Initial view in Terrain Sandcastle set to Mt Everest

### PR DESCRIPTION
**Pull request by abwood**
Minor cleanup in the Sandcastle example.  Leading this example with the TerrainProvider setup (removed local variable initialization), followed by camera setup to view Mt. Everest.

### CODE REVIEW

File: Apps/Sandcastle/gallery/Terrain.html [hunk 1]
```diff
@@ -98,7 +99,7 @@
 Sandcastle.addToolbarMenu([{
     text : 'Mount Everest',
     onselect : function() {
-        if (scene.mode === Cesium.SceneMode.SCENE3D || scene.mode === Cesium.SceneMode.COLUMBUS_VIEW) {
+        if (viewer.scene.mode === Cesium.SceneMode.SCENE3D || viewer.scene.mode === Cesium.SceneMode.COLUMBUS_VIEW) {
```
Comment (mramato) [inline Apps/Sandcastle/gallery/Terrain.html]: We actually don't need these `if` checks anymore. `lookAt` works in all camera modes.  We should remove the if checks and the else blocks.

File: Apps/Sandcastle/gallery/Terrain.html [hunk 2]
```diff
@@ -136,6 +137,7 @@
 var terrainSamplePositions;
 
 function sampleTerrainSuccess() {
+    viewer.scene.globe.depthTestAgainstTerrain = true;
```
Comment (mramato) [inline Apps/Sandcastle/gallery/Terrain.html]: Shouldn't we just set this once in the beginning immediately after Viewer construction?  I would also add a comment explaining what it is and why we need it because it's a common question on the mailing list.

File: Apps/Sandcastle/gallery/Terrain.html [hunk 3]
```diff
@@ -34,16 +34,19 @@
...
-globe.depthTestAgainstTerrain = true;
-
 var cesiumTerrainProviderMeshes = new Cesium.CesiumTerrainProvider({
     url : '//assets.agi.com/stk-terrain/world',
     requestWaterMask : true,
     requestVertexNormals : true
 });
+viewer.terrainProvider = cesiumTerrainProviderMeshes;
+
+var mountEverest = new Cesium.Cartesian3(300770.50872389384, 5634912.131394585, 2978152.2865545116);
```
Comment (mramato) [inline Apps/Sandcastle/gallery/Terrain.html]: Duplicating the initial view code like this is ugly.  The reason it was broken with previous changes is because Sandcastle currently expects the first button of the first menu to be the default option.  If you want to give me access to your fork, I can tweak the Sandcastle code to avoid having this here.  Otherwise I can just do it after we merge this into master.

File: Apps/Sandcastle/gallery/Terrain.html (hunk 1 above)
Comment (abwood) [inline Apps/Sandcastle/gallery/Terrain.html]: Didn't know that.  I'll update.

File: Apps/Sandcastle/gallery/Terrain.html (hunk 2 above)
Comment (abwood) [inline Apps/Sandcastle/gallery/Terrain.html]: The depth test is only necessary in this example when we add the billboards to the scene.  Putting it up top disconnected it from its use and also made it look like it was a required part of Terrain setup.

File: Apps/Sandcastle/gallery/Terrain.html (hunk 3 above)
Comment (abwood) [inline Apps/Sandcastle/gallery/Terrain.html]: Yep, agreed that duplicating view code is ugly.  Wrapping it in a function didn't feel right either in a sandcastle example, nor did adding an entry for viewing the entire globe (there's already a home button for that).  My only justification was that this is a sandcastle example, that is exposing code snippets for developer to experimentation, without having to wade through custom setup code.

Long story short, I'll stop down and we can talk about how this can be fixed in sandcastle.

**Comment by pjcozzi** [comment]
This is for #2489.  @mramato can merge when ready.

File: Apps/Sandcastle/gallery/Terrain.html (hunk 3 above)
Comment (abwood) [inline Apps/Sandcastle/gallery/Terrain.html]: @mramato You have access to the fork if you'd like to add the sandcastle changes.

**Comment by pjcozzi** [comment]
Can we merge this today for 1.7?

**Comment by mramato** [comment]
It's on my TODO.

**Comment by mramato** [comment]
Thanks @abwood turns out that Sandcastle change was easier than I thought it would be.
